core/src/main/java/org/apache/brooklyn/entity/group/DynamicClusterImpl.java

```diff
@@ -799,6 +795,12 @@ protected Entity replaceMember(Entity member, @Nullable Location memberLoc, Map<
     /** <strong>Note</strong> for sub-classes; this method can be called while synchronized on {@link #mutex}. */
     protected Collection<Entity> grow(int delta) {
         Preconditions.checkArgument(delta > 0, "Must call grow with positive delta.");
+        Integer maxSize = config().get(MAX_SIZE);
+        final int desiredSize = getCurrentSize() + delta;
+        if (maxSize != null && desiredSize > maxSize) {
```

Comment:
I think we need a corresponding check for `MIN_SIZE` in `shrink(int delta)`

Reply:
Which `MIN_SIZE` are you referring to? You only added the max in https://github.com/apache/brooklyn-server/pull/803.